src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,107 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.AddCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.PrintListCommand;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+
+import duke.exception.EmptyTaskDukeException;
+import duke.exception.InvalidInputDukeException;
+import duke.exception.InvalidTaskDukeException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * Represents a Parser used to read user input in Duke.
+ */
+public class Parser {
+    /**
+     * Parses input and returns a Command based on input.
+     * @param fullCommand String of user input.
+     * @return Command based on user input.
+     * @throws InvalidInputDukeException If parser does not understand user input.
+     * @throws EmptyTaskDukeException If user did not input task name.
+     * @throws InvalidTaskDukeException If user did not input appropriate DateTime for Event and Deadline.
+     */
+    public static Command parse(String fullCommand) throws InvalidInputDukeException, EmptyTaskDukeException, InvalidTaskDukeException {
+        Scanner scanner = new Scanner(fullCommand);
+        if (scanner.hasNext()) {
+            String toProcess = scanner.next();
+            switch (toProcess) {
+            case "list":
+                return new PrintListCommand();
+            case "bye":
+                return new ExitCommand();
+            case "done":
+                return new DoneCommand(scanner.nextInt());
+            case "delete":
+                return new DeleteCommand(scanner.nextInt());
+            case "find":
+                return new FindCommand(scanner.next());
+            case "todo":
+            case "deadline":
+            case "event":
+                return createAddCommand(fullCommand);
+            default:
+                throw new InvalidInputDukeException();
+            }
+        } else {
+            throw new InvalidInputDukeException();
+        }
+    }
+
+    /**
+     * Abstraction used to create AddCommand from given String.
```

Comment:
I think it's better to start the first line of the java docs comment with a verb instead?